joomla/joomla-cms
# [4.0] Correct database connection encryption fields order in the installation form

Pull Request for new issue.

### Summary of Changes

This Pull Request (PR) corrects the ordering of the fields for database connection encryption options in the installation form so it is the same as in the server tab of global configuration in backend when value"Two-way authentication" is selected in drop down "Connection Encryption". For the other values "Default (server controlled)" and "One-way authentication" of this drop down ordering is already ok.

The reason for the wrong ordering is a mistake I've made in my PR #27348 .

The form xml file [https://github.com/joomla/joomla-cms/blob/4.0-dev/installation/forms/setup.xml](https://github.com/joomla/joomla-cms/blob/4.0-dev/installation/forms/setup.xml) already has the right ordering, but the installation form doesn't use the xml to create the fields.

### Testing Instructions

1. On an installed 4.0-dev or recent nightly build, login to backend and go to the "Server" tab of "Global Configuration".
2. Select option "Two-way authentication" in the drop down "Connection Encryption".
3. Note the order of the fields for database encryption e.g. by making a screenshot.
4. Set the switcher "Server Certificate Verification" to "Yes".
5. Note again the order of the fields for database encryption e.g. by making a screenshot.
6. Start a new installation of the same 4.0-dev code or nightly build package on another server, or use the same server and delete configuration.php before starting the installation.
7. When coming to the databae connection options, select option "Two-way authentication" in the drop down "Connection Encryption".
8. Note the order of the fields for database encryption e.g. by making a screenshot.
9. Set the switcher "Server Certificate Verification" to "Yes".
10. Note again the order of the fields for database encryption e.g. by making a screenshot.
11. Compare the results from global configuration and the installation form.
**Result:** See section "Actual result" below.
12. Apply this PR.
13. Repeat steps 6 to 11.
**Result:** See section "Expected result" below.

### Expected result

The ordering of fields for database connection encryption options is the same as in global configuration in backend.
- 2-way authentication without server certificate verification:
![db-ssl-options_install_2-way-no-verify_corrected](https://user-images.githubusercontent.com/7413183/73490681-4be62a80-43ad-11ea-852f-96017142a39e.png)
- 2-way authentication with server certificate verification:
![db-ssl-options_install_2-way-verify_corrected](https://user-images.githubusercontent.com/7413183/73490695-53a5cf00-43ad-11ea-8824-ddd17cd70974.png)

### Actual result

- 2-way authentication without server certificate verification:
![db-ssl-options_install_2-way-no-verify](https://user-images.githubusercontent.com/7413183/73486958-5e109a80-43a6-11ea-986c-26844df59c57.png)
- 2-way authentication with server certificate verification:
![db-ssl-options_install_2-way-verify](https://user-images.githubusercontent.com/7413183/73490716-5d2f3700-43ad-11ea-9409-a331b0cf4873.png)

### Documentation Changes Required

No. Help screens for installation form have to be updated anyway.
